wolfcrypt/test/test.c

```diff
@@ -9073,6 +9085,103 @@ int certext_test(void)
 }
 #endif /* WOLFSSL_CERT_EXT && WOLFSSL_TEST_CERT */
 
+#if defined(WOLFSSL_DECERT_CACHE) && defined(WOLFSSL_TEST_CERT) && \
+    defined(WOLFSSL_CERT_EXT) && defined(WOLFSSL_CERT_GEN)
+int decodedCertCache_test(void)
+{
+    int ret;
+    Cert cert;
+    FILE* file;
+    byte* der;
+    word32 derSz;
+
+    derSz = FOURK_BUF;
+    der = XMALLOC(FOURK_BUF, HEAP_HINT, DYNAMIC_TYPE_TMP_BUFFER);
+    if (der == NULL)
+        return -1;
+
+    /* load cert.der */
+    file = fopen(certDerFile, "rb");
+    if (file != NULL) {
+        derSz = fread(der, 1, FOURK_BUF, file);
+        fclose(file);
+    }
+    else
+        return -1;
+
+    if (wc_InitCert(&cert)) {
+        return -1;
+    }
+
+    ret = wc_SetSubjectBuffer(&cert, der, derSz);
+    if(ret != 0)
+        return ret;
+
+    ret = wc_SetSubjectBuffer(NULL, der, derSz);
+    if(ret != BAD_FUNC_ARG)
+        return ret;
+
+    ret = wc_SetSubjectRaw(&cert, der, derSz);
+    if(ret != 0)
+        return ret;
+
+    ret = wc_SetSubjectRaw(NULL, der, derSz);
+    if(ret != BAD_FUNC_ARG)
+        return ret;
+
+    ret = wc_SetIssuerBuffer(&cert, der, derSz);
+    if(ret != 0)
+        return ret;
+
+    ret = wc_SetIssuerBuffer(NULL, der, derSz);
+    if(ret != BAD_FUNC_ARG)
+        return ret;
+
+    ret = wc_SetIssuerRaw(&cert, der, derSz);
+    if(ret != 0)
+        return ret;
+
+    ret = wc_SetIssuerRaw(NULL, der, derSz);
+    if(ret != BAD_FUNC_ARG)
+        return ret;
+
+#ifdef WOLFSSL_ALT_NAMES
+    ret = wc_SetAltNamesBuffer(&cert, der, derSz);
+    if(ret != 0)
+        return ret;
+
+    ret = wc_SetAltNamesBuffer(NULL, der, derSz);
+    if(ret != BAD_FUNC_ARG)
+        return ret;
+
+    ret = wc_SetDatesBuffer(&cert, der, derSz);
+    if(ret != 0)
+        return ret;
+
+    ret = wc_SetDatesBuffer(NULL, der, derSz);
+    if(ret != BAD_FUNC_ARG)
+        return ret;
+#endif
+
+    ret = wc_SetAuthKeyIdFromCert(&cert, der, derSz);
+    if(ret != 0)
+        return ret;
+
+    ret = wc_SetAuthKeyIdFromCert(NULL, der, derSz);
+    if(ret != BAD_FUNC_ARG)
+        return ret;
+
+    wc_DecodedCert_Cache_Free(&cert);
```

Comment:
Recommend adding `#ifndef WOLFSSL_DECERT_CACHE` around this...

Reply:
That conditional already wraps the whole test @ ln 9088